### PR TITLE
Feature/guava implementation

### PR DESCRIPTION
**Pull request by JaimLus**
## Summary
- Guava added

## Trello Card
https://trello.com/c/BCzkmkj7/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -68,7 +68,81 @@
...
+
+        checkNotNull(year, "Please check year field, its null");
+
+        this.year = year;
+
+    }
+
+    public void setPages(int pages) {
+
+        checkNotNull(pages, "Please check pages field, its null");
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java]: Careful, int is not an object, hence it cannot be null 😅

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -68,7 +68,81 @@
...
+        checkArgument(pages > 20, "Please check pages field, its can't be 20 or less than 20");
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java]: A good practice is to set constants that represent these seemingly random values, for example `MINIMUM_AMOUNT_OF_PAGES = 20`, this way you improve readability 😄

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Nice job! A few comments and it should be good to go 😄

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (JaimLus) [reply src/main/java/wolox/training/models/Book.java]: ok, what is called magic numbers.

**Review by Jimenarosello — APPROVED** [review]
Nice job! 🚀

**Review by marcegl91 — APPROVED** [review]
Great!

**Pull request merged**